Automattic/wp-calypso
# Theme Tiers: Use the Theme Filters Store

Related to https://github.com/Automattic/wp-calypso/issues/89056
Requires D145830-code

So far, we have returned the Theme Tiers definition with the `/themes` endpoint. Since themes are heavily cached in Calypso, sometimes the Tiers are not populated quickly enough.

This could be solved in many ways, but I've noticed the existing `/theme-filters` endpoint already satisfy all our needs. In D145830-code I've updated it to return the complete Tiers definitions.

This PR removes the `themes.themeTiers` store and makes Tiers use the pre-existing `themes.themeFilters.tier` instead.

## Testing Instructions

1. Apply D145830-code and sandbox the API.
2. Clear Calypso indexedDB.
3. Load the Theme Showcase.
4. Ensure the pricing dropdown is populated.
5. Check the state in console (it might be needed to reload the page with a `?debug` param.
   - `state.themes.themeTiers` should not exist anymore.
   - `state.themes.themeFilters.tier` should contain all the Tiers definition.
